### ISSUE TITLE
copy.deepcopy breaks comparability of REQUIRED, OPTIONAL, and MISSING.

### ISSUE THREAD

**Issue by mrshannon** [mrshannon/dataclass-builder#12]
Because REQUIRED, OPTIONAL, and MISSING are intended to be sentinal objects they should still be comparable after a copy.deepcopy the same way None is.

Setup
-------

```python
from copy import deepcopy
from dataclass_builder import REQUIRED, OPTIONAL, MISSING
```

Result
--------

```python
>>>None is None
True
>>>deepcopy(None) is None
True
>>>REQUIRED is REQUIRED
True
>>>deepcopy(REQUIRED) is REQUIRED
False
```

Expected Result
---------------------

```python
>>>REQUIRED is REQUIRED
True
>>>REQUIRED is deepcopy(REQUIRED)
True
```

**Comment by mrshannon** [comment]
copy.copy also breaks comparability as well.

**Comment by mrshannon** [comment]
This was based on the MISSING value from CPython's dataclass module so there might be a similar bug in that module as well.

**Comment by mrshannon** [comment]
Closed by #12
